portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,36 +23,70 @@
 import java.util.List;
 import com.google.gson.Gson;
 import java.util.Arrays;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.*;
+import java.text.*;
+import java.sql.Timestamp;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;  
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private ArrayList<String> messages;
-
-    @Override
-    public void init() {
-        messages= new ArrayList<String>();
-    }
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+Query query = new Query("Task");
+System.out.println("query:"+query);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+System.out.println("d:"+datastore);
+        PreparedQuery results = datastore.prepare(query);
+System.out.println("r:"+results);
+        List<Comment> comments = new ArrayList<>();
         
-        //converting ArrayList to json string 
-        Gson gson = new Gson();
-        String json = gson.toJson(messages);
+        for (Entity entity : results.asIterable()) {
+            Key id = entity.getKey();
+System.out.println("key:"+entity.getKey());            
+            String title = (String) entity.getProperty("title");
+System.out.println("id:"+id+" title:"+title);
+            long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
We should keep the frontend and backend consistent. Ideally we should either use "timestamp" or "date". If this is user facing, we should use "Date" which is more user friendly.

Reply:
Thanks for the info! I'll make sure to do that.